src/main/groovy/org/mockito/release/internal/gradle/E2ETestingPlugin.java

```diff
@@ -0,0 +1,100 @@
+package org.mockito.release.internal.gradle;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+
+import java.io.File;
+
+import static java.util.Arrays.asList;
+import static org.mockito.release.internal.gradle.util.StringUtil.capitalize;
+
+/**
+ * This plugin tests your library end-to-end (e2e) using client projects.
+ * Plugin clones client projects to '$buildDir/project-name-pristine' first, next clone project from 'pristine' to
+ * '$buildDir/project-name-work' and execute 'testRelease' task using the newest mockito-release-tools version
+ *
+ * Adds tasks:
+ * <ul>
+ *     <li>cloneProjectFromGitHub$projectName - {@link CloneGitRepositoryTask}</li>
+ *     <li>cloneProjectToWorkDir$projectName - {@link CloneGitRepositoryTask}</li>
+ *     <li>runTest$projectName - {@link RunTestReleaseTask}</li>
+ * </ul>
+ */
+public class E2ETestingPlugin implements Plugin<Project> {
```

Comment:
Very nice Javadoc, thank you! Makes it easy to review the code if I understand the design upfront!